src/providers/google_live.py

```
@@ -1965,24 +2036,59 @@ async def _handle_go_away(self, data: Dict[str, Any]) -> None:
                 # The Live API docs require `realtimeInput` messages to have a valid payload; sending
                 # `{ "realtimeInput": {} }` can be treated as an unsupported operation (observed as
                 # 1008 close + 501 NotImplemented in dashboards).
                 if self._setup_complete and self.websocket:
+                    # If we're actively streaming audio, don't send pings. Some accounts appear to
+                    # close connections (1008) after repeated ping frames even when audio is flowing.
+                    try:
+                        last_audio = self._last_realtime_input_sent_monotonic
+                        if last_audio is not None and (time.monotonic() - last_audio) < idle_sec:
+                            continue
+                    except Exception:
+                        pass
```

Ruff (0.15.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)